src/util/src/mem/mod.rs

```diff
@@ -4,10 +4,12 @@ use std::mem::{size_of, MaybeUninit};
 mod macros;
 
 /// Just a shortcut to `MaybeUninit::uninit().assume_init()`.
-pub fn uninit<T>() -> T {
+#[allow(clippy::uninit_assumed_init)]
+pub unsafe fn uninit<T>() -> T {
     unsafe { MaybeUninit::uninit().assume_init() }
 }
 
+#[allow(clippy::uninit_vec)]
 pub fn new_buffer<T>(size: usize) -> Vec<T>
```

Comment:
We should mark this one as unsafe too because it can cause undefined behavior if not using correctly.